agent/ssm/ssm.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
Any reason we don't have this inside `agent/taskresource` like `agent/taskresource/ssmsecret/ssm`? 
I don't see any use cases where agent would use `ssm` in a standalone way.

Reply:
What if later on, private registry or other features also need ssm?

Reply:
Yeah that could be a possibility. 
How about moving it under `agent/taskresource/` then? 

It does not feel right having it directly inside agent package now, since it's not the one using it.

Reply:
> It is a little bit weird for me that all other aws services packages are under agent folder 

you're right. historically we've kept service packages under the `agent`, i agree wth you and im inclined to leave it this way for now with `ssm` as well. 

maybe we can through everything under a `agent/awsservices` pkg? but that's refactoring work that can wait.

Reply:
sometimes our existing code base is not the best example :) we can leave this as is and refactor later when needed.